addons/purchase_stock/models/purchase_order_line.py

```diff
@@ -340,7 +340,7 @@ def _prepare_purchase_order_line_from_procurement(self, product_id, product_qty,
         res['date_planned'] = values.get('date_planned')
         # The date must be day before or equal at the supplier target day
         if po.partner_id.group_rfq == 'week' and po.partner_id.group_on != 'default':
-            delta_days = (res['date_planned'].isoweekday() - int(po.partner_id.group_on)) % 7
+            delta_days = (res['date_planned'].isoweekday() + 1 - int(po.partner_id.group_on)) % 7
```

Comment:
Not sure I understand. Let's say the `date_planned` is on a Monday and partner's `group_on` as well:
This would lead to a `delta_days = (1 + 1 - 1) % 7 = 1`
So the `date_planned` would be set one day in the past in the following line?